### PR TITLE
Mungegithub upgrade 

### PR DESCRIPTION
**Pull request by yutongz**
**Release note**:

```release-note
none
```
Based on the feedback of the survey we sent out before, we did some functionality changes to make life easier.

This PR documents the main changes from upgrading mungegithub
* "do-not-merge" becomes a [label family](https://github.com/yutongz/test-infra/tree/mungegithub_update/mungegithub#labels): general "do-not-merge", "do-not-merge/hold" and "do-not-merge/release-note-label-needed"
  * _[New feature, backwards compatible]_ We can comment "/hold", and "do-not-merge/hold" label will be added to block merge. To unblock, comment "/hold cancel"
  * _[New feature, no behavior change]_ "do-not-merge/release-note-label-needed" label will be added due to the missing release-note message (can be string "none") and will be **automatically removed** after you add something in release-note part
* _[Removed feature, NOT backwards compatible]_ You **cannot** comment "/release-note-none" to bypass check, the only way you can do now is writing "none" in release-note section in PR description
* _[New feature, backwards compatible]_ You can now put **"action required"** in release-note section to bypass check temporarily and give you a chance to go back and add things later on (before release).

Also we fixed some [bugs](https://github.com/yutongz/test-infra/tree/mungegithub_update/mungegithub/deployment#troubleshooting):
* Submit Queue complains about CI status is not green though it is. (webhook issue)
* Prow misses "/lgtm" from time to time. (webhook event issue)

View rich diff: https://github.com/istio/test-infra/pull/542/files?short_path=617180c#diff-617180c40c1cb616cab19e763acd538a

### CODE REVIEW

**Review by ldemailly — APPROVED** [review]
thank you

File: mungegithub/README.md
Comment (andraxylia) [inline mungegithub/README.md]: feather?

Comment (andraxylia) [reply mungegithub/README.md]: s/is going/will

File: mungegithub/README.md
Comment (andraxylia) [inline mungegithub/README.md]: It's an easy way for anybody to hold the merge; if you simply comment, the robot will add and will block... use same verb tense

How does it unblock?

File: mungegithub/README.md
Comment (andraxylia) [reply mungegithub/README.md]: when PRs are created - correct tense, also capitalize PRs everywhere

File: mungegithub/deployment/README.md
Comment (andraxylia) [inline mungegithub/deployment/README.md:44]: cool, I did not know about this.

File: mungegithub/README.md
Comment (yutongz) [reply mungegithub/README.md]: Thanks for pointing out these :)

File: mungegithub/README.md
Comment (yutongz) [reply mungegithub/README.md]: By comment "/hold cancel", Prow will remove that label and the PR is good to go.

File: mungegithub/README.md
Comment (andraxylia) [inline mungegithub/README.md]: s/reuqired/required

It waits for all required tests to pass, and ensure PR has these four labels added - cla, lgtm, approve, release-note - before putting this PR into the queue. When a mergeable PR comes to the head of Submit-queue, it will trigger required retest contexts, to make sure everything is well tested. After these second-round tests pass, Mungegithub will go and merge this PR for you!

File: mungegithub/README.md
Comment (andraxylia) [inline mungegithub/README.md]: s/verities/verifies.

File: mungegithub/README.md
Comment (andraxylia) [inline mungegithub/README.md]: This contradicts what is below. Do you mean : why my PR is not merged by the bot?
- missing one of the four labels - ...-
- has "do-not-merge" label
- tests failed.

Comment (wattli) [reply mungegithub/README.md]: Retest means rerun pre-submit, if i remember correctly.

File: mungegithub/deployment/README.md
Comment (wattli) [inline mungegithub/deployment/README.md:47]: We have more repos in istio, what about others? e.g. istio/api

File: mungegithub/README.md
Comment (yutongz) [reply mungegithub/README.md]: Yeah, you guys are both right. First people are wondering why my PR isn't merged, while technically, in many case, it hasn't been re-test because of some following reasons. But, right, I am going to make it clearer.

File: mungegithub/deployment/README.md
Comment (yutongz) [reply mungegithub/deployment/README.md:47]: Right, we haven't cover all repos because usually they don't have as many PRs as those main repos. And like "api", we don't do auto-dependency-update, so it's not that necessary. But in unifying experience perspective, we should.

**Pull request merged**
